pootle/apps/pootle_store/migrations/0055_fill_unit_source_data.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10.5 on 2017-04-28 11:31
+from __future__ import unicode_literals
+
+import logging
+
+from hashlib import md5
+
+from django.conf import settings
+from django.db import migrations
+from django.utils.encoding import force_bytes
+
+from bulk_update.helper import bulk_update
+
+from pootle_misc.util import import_func
+from pootle_store.fields import to_python
+from pootle_store.utils import UnitWordcount
+
+
+logger = logging.getLogger(__name__)
+
+
+def set_unit_source_data(apps, schema_editor):
+    stores_qs = apps.get_model("pootle_store.Store").objects.all()
+    unit_sources_qs = apps.get_model(
+        "pootle_store.UnitSource").objects.select_related('unit')
+    counter = UnitWordcount(import_func(settings.POOTLE_WORDCOUNT_FUNC))
+    for store in stores_qs.iterator():
+        unit_sources = list(unit_sources_qs.filter(unit__store=store,
+                                                   source_wordcount=0))
+        for s in unit_sources:
+            s.source_hash = md5(force_bytes(s.unit.source_f)).hexdigest()
+            s.source_length = len(s.unit.source_f)
+            s.source_wordcount = max(
+                1, (counter.count_words(to_python(s.unit.source_f)) or 0))
```

Comment:
We have units with blank string or with just whitespace, so I don't think we want the `max(1, ...)` here.